File: solutions/cpp/raindrops/1/raindrops.cpp
#include "raindrops.h"

namespace raindrops {
std::string convert(int number) {
        std::string str1 = "";
        std::string str2 = "";
        std::string str3{""};
        
        if (number % 3 == 0) {
            str1 = "Pling";
        }
        if (number % 5 == 0) {
            str2 = "Plang";
        }
        if (number % 7 == 0) {
            str3 = "Plong";
        }
        if (str1 == str2 and str2 == str3) {
            return std::to_string(number);
        }
        return str1 + str2 + str3;

    }

}  // namespace raindrops

File: solutions/r/raindrops/1/raindrops.R
raindrops <- function(number) {
  a1 = number %% 3
  a2 = number %% 5
  a3 = number %% 7
  result = ''
  if ( a1 * a2 * a3 != 0) {
    paste0(number)
    } else {
  if (a1 == 0) {result = paste0(result, 'Pling')}
  if (a2 == 0) {result = paste0(result, 'Plang')}
  if (a3 == 0) {result = paste0(result, 'Plong')}
   result }
}
